src/battle_script_commands.c

```diff
@@ -2934,6 +2939,7 @@ void SetMoveEffect(bool32 primary, u32 certain)
         INCREMENT_RESET_RETURN
 
     if (gBattleMons[gEffectBattler].hp == 0
+        && !activateAfterFaint
```

Comment:
Is the variable needed? You could just do the same as with the already existing move effects here.

gBattleScripting.moveEffect != MOVE_EFFECT_STEALTH_ROCK && gBattleScripting.moveEffect != MOVE_EFFECT_SPIKES

Reply:
How so?

The variable is used to tell the code that the effects should activate even if the opponent faints.

I added it because they weren't.

Reply:
Alright, I see what you mean now. Sorry, I didn't even bother to read the code earlier in the morning.

Though wouldn't it be better to keep the variable and instead, incorporate `MOVE_EFFECT_PAYDAY`, `MOVE_EFFECT_STEAL_ITEM` and `MOVE_EFFECT_BUG_BITE` into the `switch (gBattleScripting.moveEffect)`? 🤔 

Kinda feels things would be more readable like that.

E.g.
```diff
diff --git a/src/battle_script_commands.c b/src/battle_script_commands.c
index d0de44b72..336655b0e 100644
--- a/src/battle_script_commands.c
+++ b/src/battle_script_commands.c
@@ -2896,6 +2896,9 @@ void SetMoveEffect(bool32 primary, u32 certain)
         return;
     case MOVE_EFFECT_STEALTH_ROCK:
     case MOVE_EFFECT_SPIKES:
+    case MOVE_EFFECT_PAYDAY:
+    case MOVE_EFFECT_STEAL_ITEM:
+    case MOVE_EFFECT_BUG_BITE:
         activateAfterFaint = TRUE;
         break;
     }
@@ -2938,11 +2941,7 @@ void SetMoveEffect(bool32 primary, u32 certain)
     if (TestSheerForceFlag(gBattlerAttacker, gCurrentMove) && affectsUser != MOVE_EFFECT_AFFECTS_USER)
         INCREMENT_RESET_RETURN
 
-    if (gBattleMons[gEffectBattler].hp == 0
-        && !activateAfterFaint
-        && gBattleScripting.moveEffect != MOVE_EFFECT_PAYDAY
-        && gBattleScripting.moveEffect != MOVE_EFFECT_STEAL_ITEM
-        && gBattleScripting.moveEffect != MOVE_EFFECT_BUG_BITE)
+    if (gBattleMons[gEffectBattler].hp == 0 && !activateAfterFaint)
         INCREMENT_RESET_RETURN
 
     if (DoesSubstituteBlockMove(gBattlerAttacker, gEffectBattler, gCurrentMove) && affectsUser != MOVE_EFFECT_AFFECTS_USER)
```

Reply:
I could rename `activateAfterFaint` to `activateAfterTargetFaints` for extra readability too.

Reply:
Pinging @DizzyEggg to get a response 😅 ...